### PR TITLE
Feature/KN-7731

### PR DESCRIPTION
**Pull request by igor-kulik**
Replacement products info retrieval functionality of SwiftyStoreKit by own implementation

### CODE REVIEW

File: PurchaseController/Classes/PurchaseController.swift [hunk 1]
```diff
@@ -78,11 +78,21 @@ public final class PurchaseController: PaymentQueueControllerDelegate {
         self?.purchaseActionState = .finish(PurchaseActionResult.error(error.purchaseError))
     }
     
+    // MARK: - ProductsInfoObserver
+    
+    lazy var onRetrieve: ((RetrievedProductsInfo) -> Void)? = { [weak self] (retrievedProductsInfo)  in
+            self?.persistor.persist(products: retrievedProductsInfo.products)
+        let hasInvalidProducts = retrievedProductsInfo.invalidProductIdentifiers.count > 0
```
Comment (ValeriyJefimov) [inline PurchaseController/Classes/PurchaseController.swift]: can we use isEmpty?

File: PurchaseController/Classes/PurchaseController.swift [hunk 2]
```diff
@@ -78,11 +78,21 @@ public final class PurchaseController: PaymentQueueControllerDelegate {
         self?.purchaseActionState = .finish(PurchaseActionResult.error(error.purchaseError))
     }
     
+    // MARK: - ProductsInfoObserver
+    
+    lazy var onRetrieve: ((RetrievedProductsInfo) -> Void)? = { [weak self] (retrievedProductsInfo)  in
+            self?.persistor.persist(products: retrievedProductsInfo.products)
```
Comment (ValeriyJefimov) [inline PurchaseController/Classes/PurchaseController.swift]: remove whitespacing

File: PurchaseController/Classes/ProductsInfoController.swift [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    var onRetrieve: ((_ productsInfo: RetrievedProductsInfo) -> Void)? { get set }
+    var onError: ((_ error: Error) -> Void)? { get set }
+}
+
+final class PCProductsInfoController: NSObject {
+    
+    // MARK: - Properties
+
+    private weak var observer: ProductsInfoObserver?
+    private var productsRequest: SKProductsRequest?
```
Comment (ValeriyJefimov) [inline PurchaseController/Classes/ProductsInfoController.swift:23]: can it be weak?

Comment (igor-kulik) [reply PurchaseController/Classes/ProductsInfoController.swift:23]: No, as per our latest discussion, same case also included in https://github.com/dashdevs/PurchaseController/pull/8

**Review by ValeriyJefimov — APPROVED** [review]

**Pull request merged**